eureka/src/main/java/com/linecorp/armeria/client/eureka/EurekaEndpointGroup.java

```diff
@@ -181,45 +183,60 @@ public static EurekaEndpointGroupBuilder builder(
     }
 
     private void fetchRegistry() {
+        if (closed) {
+            return;
+        }
         final HttpResponse response;
         final ClientRequestContext ctx;
-        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
-            response = webClient.execute(requestHeaders);
-            ctx = captor.get();
-        }
-
-        final EventLoop eventLoop = ctx.eventLoop().withoutContext();
-        response.aggregateWithPooledObjects(eventLoop, ctx.alloc()).handle((aggregatedRes, cause) -> {
-            try (HttpData content = aggregatedRes.content()) {
+        EventLoop eventLoop0 = null;
+        try {
+            try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+                response = webClient.execute(requestHeaders);
+                ctx = captor.get();
+            }
+            EventLoop eventLoop = ctx.eventLoop().withoutContext();
+            eventLoop0 = eventLoop;
+            response.aggregateWithPooledObjects(eventLoop, ctx.alloc()).handle((aggregatedRes, cause) -> {
                 if (closed) {
                     return null;
                 }
                 if (cause != null) {
                     logger.warn("Unexpected exception while fetching the registry from: {}." +
                                 " (requestHeaders: {})", webClient.uri(), requestHeaders, cause);
+                } else if (aggregatedRes == null) {
+                    logger.warn("Got null response while fetching the registry from: {}.", webClient.uri());
```

Comment:
Should we remove this branch? Because `aggregateWithPooledObjects` never returns a null response if the `cause` is null.